### PR TITLE
Restore module

### PR DESCRIPTION
**Pull request by t-anssingh102**
- Generated backup cmdlets
- added egs and renamed command
- Fixed names and argument description
- Added command to get default policy template Added customization for New-BackupPolicy command
- parameters added to modify retention policy
- Edit-SchedulePolicy
- saphana working with new params

## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by t-anssingh102** [comment]
I'll be adding the examples etc too

File: src/RecoveryServices/RecoveryServices.Backup.Autorest/Az.RecoveryServices.sln [hunk 1]
```diff
@@ -0,0 +1,25 @@
+﻿
+Microsoft Visual Studio Solution File, Format Version 12.00
+# Visual Studio Version 17
+VisualStudioVersion = 17.6.33801.468
+MinimumVisualStudioVersion = 10.0.40219.1
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Az.RecoveryServices", "Az.RecoveryServices.csproj", "{763EC316-4266-4BFB-A330-BE4E6CA365F1}"
```
Comment (hiaga) [inline src/RecoveryServices/RecoveryServices.Backup.Autorest/Az.RecoveryServices.sln:6]: remove this file

File: src/RecoveryServices/RecoveryServices.Backup.Autorest/custom/Cmdlets/Restore/AzureVM-createNew-newVM.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
+{
+    "requests": [
+        {
+            "content": {
+                "properties": {
+                    "objectType": "IaasVMRestoreRequest",
```
Comment (hiaga) [inline src/RecoveryServices/RecoveryServices.Backup.Autorest/custom/Cmdlets/Restore/AzureVM-createNew-newVM.json]: can you share these files on teams and remove from repo now ? I see 4 such files

File: src/RecoveryServices/RecoveryServices.Backup.Autorest/custom/Cmdlets/Restore/SAPHANA-alternateLocation-pointInTime.json [hunk 3]
```diff
@@ -0,0 +1,27 @@
+{
+    "requests": [
+        {
+            "content": {
+                "properties": {
+                    "objectType": "AzureWorkloadSAPHanaPointInTimeRestoreRequest",
+                    "targetInfo": {
+                        "containerId": "/subscriptions/38304e13-357e-405e-9e9a-220351dcce8c/resourceGroups/saphana-eus-pstest-rg/providers/Microsoft.RecoveryServices/vaults/saphana-eus-pstest-vault/backupFabrics/Azure/protectionContainers/VMAppContainer;Compute;saphana-eus-pstest-rg;saphana-eus-pstest-vm",
+                        "databaseName": "ARV/arv11",
+                        "overwriteOption": "FailOnConflict"
```
Comment (hiaga) [inline src/RecoveryServices/RecoveryServices.Backup.Autorest/custom/Cmdlets/Restore/SAPHANA-alternateLocation-pointInTime.json]: same for SAPHANA and SQL jsons as well. create a folder with all these requests and share the folder as a zip

File: src/RecoveryServices/RecoveryServices.Backup.Autorest/custom/Helpers/RestoreHelpers.ps1 [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+        Write-Debug -Message "Region: $Region"
+        Write-Debug -Message "NonRecoverable: $NonRecoverable"
+        Write-Debug -Message "DataSourceLogicalName: $DataSourceLogicalName"
+        Write-Debug -Message "DataSourcePath: $DataSourcePath"
+        Write-Debug -Message "DataTargetPath: $DataTargetPath"
+        Write-Debug -Message "LogSourceLogicalName: $LogSourceLogicalName"
+        Write-Debug -Message "LogSourcePath: $LogSourcePath"
+        Write-Debug -Message "LogTargetPath: $LogTargetPath"
+
+        # Validate the parameters
```
Comment (hiaga) [inline src/RecoveryServices/RecoveryServices.Backup.Autorest/custom/Helpers/RestoreHelpers.ps1:156]: what is this function for ? I don't see any code inside it

File: src/RecoveryServices/RecoveryServices.Backup.Autorest/docs/Az.RecoveryServices.md [hunk 5]
```diff
@@ -169,6 +169,12 @@ Fetches the status of a triggered validate operation.
 The status can be in progress, completed\r\nor failed.
 You can refer to the OperationStatus enum for all the possible states of the operation.\r\nIf operation has completed, this method returns the list of errors obtained while validating the operation.
 
+### [Get-SourceResourceId](Get-SourceResourceId.md)
+Initializes Restore Request object for triggering restore on a protected backup instance.
```
Comment (hiaga) [inline src/RecoveryServices/RecoveryServices.Backup.Autorest/docs/Az.RecoveryServices.md]: this function should not be exported as a command. There is an attribute for it.

File: src/RecoveryServices/RecoveryServices.Backup.Autorest/docs/Az.RecoveryServices.md [hunk 6]
```diff
@@ -196,6 +202,9 @@ To know the status of the operation, call the GetItemOperationResult API.
 Create Intent for Enabling backup of an item.
 This is a synchronous operation.
 
+### [New-AzRecoveryServicesRestoreRequest](New-AzRecoveryServicesRestoreRequest.md)
+Creates a new backup policy in a given recovery services vault
```
Comment (hiaga) [inline src/RecoveryServices/RecoveryServices.Backup.Autorest/docs/Az.RecoveryServices.md]: update description in command

File: src/RecoveryServices/RecoveryServices.Backup.Autorest/docs/Get-SourceResourceId.md [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+
+## DESCRIPTION
+Initializes Restore Request object for triggering restore on a protected backup instance.
+
+## EXAMPLES
+
+### Example 1: {{ Add title here }}
+```powershell
+{{ Add code here }}
+```
```
Comment (hiaga) [inline src/RecoveryServices/RecoveryServices.Backup.Autorest/docs/Get-SourceResourceId.md]: remove this file. this file shouldn't get generated

File: src/RecoveryServices/RecoveryServices.Backup.Autorest/docs/Initialize-AzRecoveryServicesRestoreRequest.md [hunk 8]
```diff
@@ -0,0 +1,561 @@
...
+
+## DESCRIPTION
+Initializes Restore Request object for triggering restore on a protected backup instance.
+
+## EXAMPLES
+
+### Example 1: {{ Add title here }}
+```powershell
+PS C:\> {{ Add code here }}
+
```
Comment (hiaga) [inline src/RecoveryServices/RecoveryServices.Backup.Autorest/docs/Initialize-AzRecoveryServicesRestoreRequest.md:37]: add examples and test cases

File: src/RecoveryServices/RecoveryServices.Backup.Autorest/docs/New-AzRecoveryServicesRestoreRequest.md [hunk 9]
```diff
@@ -0,0 +1,205 @@
...
+## DESCRIPTION
+Creates a new backup policy in a given recovery services vault
+
+## EXAMPLES
+
+### Example 1: {{ Add title here }}
+```powershell
+PS C:\> {{ Add code here }}
+
+{{ Add output here }}
```
Comment (hiaga) [inline src/RecoveryServices/RecoveryServices.Backup.Autorest/docs/New-AzRecoveryServicesRestoreRequest.md:30]: add examples and test cases

File: src/RecoveryServices/RecoveryServices.Backup.Autorest/examples/Get-SourceResourceId.md [hunk 10]
```diff
@@ -0,0 +1,22 @@
+### Example 1: {{ Add title here }}
+```powershell
+{{ Add code here }}
+```
+
+```output
+{{ Add output here }}
+```
+
```
Comment (hiaga) [inline src/RecoveryServices/RecoveryServices.Backup.Autorest/examples/Get-SourceResourceId.md]: delete this file

File: src/RecoveryServices/RecoveryServices.Backup.Autorest/examples/Initialize-AzRecoveryServicesRestoreRequest.md [hunk 11]
```diff
@@ -0,0 +1,18 @@
+### Example 1: {{ Add title here }}
+```powershell
+PS C:\> {{ Add code here }}
+
+{{ Add output here }}
+```
+
+{{ Add description here }}
```
Comment (hiaga) [inline src/RecoveryServices/RecoveryServices.Backup.Autorest/examples/Initialize-AzRecoveryServicesRestoreRequest.md:8]: add examples

File: src/RecoveryServices/RecoveryServices.Backup.Autorest/examples/New-AzRecoveryServicesRestoreRequest.md [hunk 12]
```diff
@@ -0,0 +1,18 @@
+### Example 1: {{ Add title here }}
+```powershell
+PS C:\> {{ Add code here }}
+
+{{ Add output here }}
+```
+
```
Comment (hiaga) [inline src/RecoveryServices/RecoveryServices.Backup.Autorest/examples/New-AzRecoveryServicesRestoreRequest.md:7]: add examples

File: src/RecoveryServices/RecoveryServices.Backup.Autorest/custom/Cmdlets/Restore/New-AzRecoveryServicesRestoreRequest.ps1 [hunk 13]
```diff
@@ -0,0 +1,108 @@
+﻿function New-AzRecoveryServicesRestoreRequest
+{
+    [OutputType('Microsoft.Azure.PowerShell.Cmdlets.RecoveryServices.Models.Api20230201.IRestoreRequest')]
+    [CmdletBinding(PositionalBinding=$false, SupportsShouldProcess)]
+    [Microsoft.Azure.PowerShell.Cmdlets.RecoveryServices.Description('Creates a new backup policy in a given recovery services vault')]
+
+	param(
+        [Parameter(Mandatory=$false, HelpMessage='Subscription Id')]
```
Comment (hiaga) [inline src/RecoveryServices/RecoveryServices.Backup.Autorest/custom/Cmdlets/Restore/New-AzRecoveryServicesRestoreRequest.ps1]: update help

File: src/RecoveryServices/RecoveryServices.Backup.Autorest/custom/Cmdlets/Restore/New-AzRecoveryServicesRestoreRequest.ps1 [hunk 14]
```diff
@@ -0,0 +1,108 @@
...
+
+        [Parameter(Mandatory, HelpMessage='The name of the resource group where the recovery services vault is present.')]
+        [System.String]
+        ${ResourceGroupName},
+
+        [Parameter(Mandatory, HelpMessage='The name of the recovery services vault.')]
+        [System.String]
+        ${VaultName},
+
+        [Parameter(Mandatory, HelpMessage='Policy Name for the policy to be created')]
```
Comment (hiaga) [inline src/RecoveryServices/RecoveryServices.Backup.Autorest/custom/Cmdlets/Restore/New-AzRecoveryServicesRestoreRequest.ps1]: updated help at all params where needed

File: src/RecoveryServices/RecoveryServices.Backup.Autorest/custom/Cmdlets/Restore/Initialize-AzRecoveryServicesRestoreRequest.ps1 [hunk 15]
```diff
@@ -0,0 +1,386 @@
+﻿function Get-SourceResourceId {
+    [OutputType('string')]
+    [CmdletBinding(PositionalBinding=$false)]
+    [Microsoft.Azure.PowerShell.Cmdlets.RecoveryServices.Description('Initializes Restore Request object for triggering restore on a protected backup instance.')]
+
+    param (
+        [Parameter(ParameterSetName='InitializeRestoreRequest', HelpMessage='Specifies the ContainerId')]
+        [Microsoft.Azure.PowerShell.Cmdlets.RecoveryServices.Models.Api20230201.IRecoveryPointResource]
+        ${RecoveryPoint}
+    )
```
Comment (hiaga) [inline src/RecoveryServices/RecoveryServices.Backup.Autorest/custom/Cmdlets/Restore/Initialize-AzRecoveryServicesRestoreRequest.ps1]: can we move this function to backup helper

File: src/RecoveryServices/RecoveryServices.Backup.Autorest/custom/Cmdlets/Restore/Initialize-AzRecoveryServicesRestoreRequest.ps1 [hunk 16]
```diff
@@ -0,0 +1,386 @@
...
+    [Microsoft.Azure.PowerShell.Cmdlets.RecoveryServices.Description('Initializes Restore Request object for triggering restore on a protected backup instance.')]
+
+	param (
+        # Recovery point and type specific parameters
+
+        [Parameter(ParameterSetName='InitializeRestoreRequest', HelpMessage='Specifies the ContainerId')]
+        [Microsoft.Azure.PowerShell.Cmdlets.RecoveryServices.Support.DatasourceTypes]
+        ${DatasourceType},
+
+        [Parameter(ParameterSetName='InitializeRestoreRequest', HelpMessage='Specifies the ContainerId')]
```
Comment (hiaga) [inline src/RecoveryServices/RecoveryServices.Backup.Autorest/custom/Cmdlets/Restore/Initialize-AzRecoveryServicesRestoreRequest.ps1]: update help text for params

File: src/RecoveryServices/RecoveryServices.Backup.Autorest/custom/Helpers/RestoreHelpers.ps1 (hunk 4 above)
Comment (t-anssingh102) [reply src/RecoveryServices/RecoveryServices.Backup.Autorest/custom/Helpers/RestoreHelpers.ps1:156]: Will add validations here

**Pull request merged**